libs/ledger-live-common/src/e2e/speculosCI.ts

```diff
@@ -0,0 +1,161 @@
+import axios from "axios";
+import {
+  conventionalAppSubpath,
+  DeviceParams,
+  reverseModelMap,
+} from "@ledgerhq/speculos-transport";
+import { SpeculosDevice } from "./speculos";
+import https from "https";
+
+const { SEED, GITHUB_TOKEN, AWS_ROLE, CLUSTER } = process.env;
+const GIT_API_URL = "https://api.github.com/repos/LedgerHQ/actions/actions/";
+const START_WORKFLOW_ID = "workflows/161487603/dispatches";
+const STOP_WORKFLOW_ID = "workflows/161487604/dispatches";
+const GITHUB_REF = "main";
+const getSpeculosAddress = (runId: string) => `https://${runId}.speculos.aws.stg.ldg-tech.com`;
+const speculosPort = 443;
+
+function uniqueId(): string {
+  const timestamp = Date.now().toString(36);
+  const randomString = Math.random().toString(36).slice(2, 7);
+  return timestamp + randomString;
+}
+
+/**
+ * Helper function to make API requests with error handling
+ */
+async function githubApiRequest<T = unknown>({
```

Comment:
[ASK] Why not use Octokit instead of axios here ?

Reply:
We're making only two simple API calls and don’t want to add another dependency. But if you have reasons to switch to Octokit, let me know 🙏